src/engine/engine.js

```diff
@@ -35,21 +35,44 @@ function createState(initState) {
   };
 }
 
+class ContextWrapper extends Component {
+  getChildContext() {
+    return { store: this.store }
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    this.store = props.store
+  }
+
+  render() {
+    return Children.only(this.props.children)
+  }
+}
+
+ContextWrapper.childContextTypes = {
+  store: React.PropTypes.any
+};
+
 export default class Engine extends Component {
   constructor(props, context) {
     super(props, context);
     this.state = createState(props.initState);
+
+    this.updateComponentState = this.updateComponentState.bind(this);
+
+    sagaMiddleware.run(watchReadPerform);
+    this.state.store.subscribe(this.updateComponentState);
   }
 
   updateComponentState() {
     const store = this.state.store;
     const newState = toImmutable(store.getState());
-    this.setState({store, initState: newState})
+    this.setState({initState: newState})
   }
 
-  componentWillMount() {
-    sagaMiddleware.run(watchReadPerform);
-     this.state.store.subscribe(this.updateComponentState.bind(this));
+  shouldComponentUpdate(nextProps, nextState) {
+    return nextState.initState !== this.state.initState;
```

Comment:
shouldComponentUpdate should always return true for the Engine. If something is triggered to update the component, then something will be changed in the atom.